@here/harp-datasource-protocol/lib/InterpolatedProperty.ts

```diff
@@ -103,3 +63,179 @@ export function isInterpolatedProperty<T>(p: any): p is InterpolatedProperty<T>
     }
     return false;
 }
+
+/**
+ * Get the value of the specified property at the given zoom level, represented as a `number` value.
+ *
+ * @param property Property of a technique.
+ * @param level Display level the property should be rendered at.
+ * @param pixelToMeters Optional pixels to meters conversion factor (needed for proper
+ * interpolation of `length` values).
+ *
+ */
+export function getPropertyValue<T>(
+    property: InterpolatedProperty<T> | MaybeInterpolatedProperty<T>,
+    level: number,
+    pixelToMeters: number = 1.0
+): number {
+    if (isInterpolatedPropertyDefinition<T>(property)) {
+        throw new Error("Cannot interpolate a InterpolatedPropertyDefinition.");
+    } else if (!isInterpolatedProperty(property)) {
+        if (typeof property !== "string") {
+            return (property as unknown) as number;
+        } else {
+            if (
+                StringEncodedNumeralFormats[StringEncodedNumeralType.Meters].regExp.test(property)
+            ) {
+                return StringEncodedNumeralFormats[StringEncodedNumeralType.Meters].decoder(
+                    property
+                )[0];
+            } else if (
+                StringEncodedNumeralFormats[StringEncodedNumeralType.Pixels].regExp.test(property)
+            ) {
+                return (
+                    StringEncodedNumeralFormats[StringEncodedNumeralType.Pixels].decoder(
+                        property
+                    )[0] * pixelToMeters
+                );
+            } else {
+                for (
+                    let i = StringEncodedNumeralType.HexRGB;
+                    i <= StringEncodedNumeralType.HSL;
+                    ++i
+                ) {
+                    if (StringEncodedNumeralFormats[i].regExp.test(property)) {
+                        const hslValues = StringEncodedNumeralFormats[i].decoder(property);
+                        const rgbValues = hslToRGB(hslValues[0], hslValues[1], hslValues[2]);
+                        let result = "0x";
+                        for (const value of rgbValues) {
+                            // tslint:disable:no-bitwise
+                            result += (
+                                "0" + ((MathUtils.clamp(value, 0, 1) * 255) | 0).toString(16)
+                            ).slice(-2);
+                        }
+                        return parseInt(result, 16);
+                    }
+                }
+            }
+            throw new Error(`No StringEncodedNumeralFormat matched ${property}.`);
+        }
+    } else if (property._stringEncodedNumeralType !== undefined) {
+        if (
```

Comment:
Shorthand:

```
if ([StringEncodedNumeralType.Meters, StringEncodedNumeralType.Pixels].includes(property._stringEncodedNumeralType))
```

Reply:
I believe than 2 comparisons and a boolean operator is way more performant than a temporary object construction and a function call. It's not terribly verbose neither, so I'm not sure if this would add much clarity...

Reply:
@amsgames94 ok then